### PR TITLE
Restructure project to maven standard

### PR DESCRIPTION
**Pull request by Narimm**
Firstly there are NO code changes in this commit - it is strictly a restructure.  It brings the repository into a maven standard. Stripping support files out of the source code and moving them the correct directory structure to ensure readability going forward.

- Move tests to the correct directory
- Create resources and move them to the correct directory
- Create .github and move templates.
- Add a default encoding to ensure cross platform building is consistent.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -32,20 +33,21 @@
...
           </testResource>
         </testResources>
         <finalName>${project.name}</finalName>
         <resources>
             <resource>
             <!-- Use plugin.yml in root directory-->
             <!-- This can also automatically update plugin.yml version from pom, if you use ${project.version} as version number-->
-                <directory>${basedir}</directory>
+                <directory>${basedir}/src/main/resources</directory>
+                <targetPath>.</targetPath>
```
Comment (RoboMWM) [inline pom.xml]: targetPath is for..?

**Comment by RoboMWM** [comment]
Seems okay, though this is something that's already done in version 20. But I guess since there's really no open PRs that are ready to be merged (with exception of the hyperconomy one I suppose, which is only a single line anyways), I guess this is okay. I'll pull when I have time to just double check things over.

**Comment by Narimm** [comment]
`<targetpath>` definitively defines where you want resources to end up in the jar...
rather than relying on default.

**Comment by Narimm** [comment]
@RoboMWM where is version 20?

**Comment by RoboMWM** [comment]
#63

**Review by RoboMWM — CHANGES_REQUESTED** [review]
So for some of these not sure what happened, if you changed line endings or something? Unsure if you can somehow make it so all files are just a move instead of some files having a random full delete and insert; otherwise looks good.

**Comment by Narimm** [comment]
Ok so I updated the branch and fixed the stupid line ending changes that intellij must make...I had to use git cmd line ....but its done!!!!

**Comment by Narimm** [comment]
RobMWM,  mananged to get the moves done without the silly LF changes (I think it was line ending s that cause the problem earlier

**Review by RoboMWM — APPROVED** [review]

**Comment by RoboMWM** [comment]
Yay, much cleaner diff c:

**Pull request merged**